lib/models/event.js

```diff
@@ -189,5 +189,28 @@ module.exports.MatrixEvent.prototype = {
 
     getUnsigned: function() {
         return this.event.unsigned || {};
-    }
+    },
+
+    /**
+     * Update the content of an event in the same way it would be by the server
+     * if it were redacted before it was sent to us
+     *
+     * @param {Object} the raw event causing the redaction
+     */
+    makeRedacted: function(redaction_event) {
+        if (!this.event.unsigned) {
+            this.event.unsigned = {};
+        }
+        this.event.unsigned.redacted_because = redaction_event;
+        this.event.content = {};
```

Comment:
It should be sufficient for the case of redacting offensive displaynames; the displayname is in the `content` key so is removed along with everything else.

There is a problem in that `m.room.member` events are used to update a cached copy of the user's metadata, which is then copied to each event they send. We don't update the cache on redactions, nor any events they have sent - which means that any such events will still show the redacted displayname, as will the member list on the RHS. However, I don't believe this is a new issue.

If anything, we are removing too much stuff from `content`. It's probably best we follow the specified algorithm here, rather than allow people to build on functionality which isn't really correct; I'll tighten this up.